infra/label_evaluator_step_functions/lambdas/discover_patterns.py

```
@@ -0,0 +1,412 @@
+
+s3 = boto3.client("s3")
+
+
+def load_json_from_s3(bucket: str, key: str) -> dict[str, Any] | None:
+    """Load JSON data from S3, return None if not found."""
+    try:
+        response = s3.get_object(Bucket=bucket, Key=key)
+        return json.loads(response["Body"].read().decode("utf-8"))
+    except s3.exceptions.NoSuchKey:
+        return None
+    except Exception as e:
```

Ruff (0.14.10)

Do not catch blind exception: `Exception`

(BLE001)